asserts/device_asserts.go

```diff
@@ -166,8 +175,27 @@ func assembleModel(assert assertionBase) (Assertion, error) {
 		return nil, err
 	}
 
-	for _, mandatory := range modelMandatory {
-		if _, err := checkNotEmptyString(assert.headers, mandatory); err != nil {
+	classic, err := checkOptionalBool(assert.headers, "classic")
+	if err != nil {
+		return nil, err
+	}
+
+	if classic {
+		_, ok := assert.headers["kernel"]
+		if ok {
+			return nil, fmt.Errorf("classic model cannot specify explicitly a kernel")
```

Comment:
This sounds slightly odd to my (non-native speaker) ears. Maybe `classic model cannot explicitly specify a kernel` or `cannot specify a kernel with a classic model` ?